src/lib/HST/Effect/Report.hs

```diff
@@ -0,0 +1,160 @@
+import           Control.Monad                  ( (>=>)
+                                                , when
+                                                )
+import           Polysemy                       ( Member
+                                                , Members
+                                                , Sem
+                                                , intercept
+                                                , interpret
+                                                , makeSem
+                                                , raise
+                                                , raiseUnder2
+                                                )
+import           Polysemy.Embed                 ( Embed
+                                                , embed
+                                                )
+import           Polysemy.Error                 ( Error
+                                                , runError
+                                                , fromExceptionSem
+                                                )
+import           Polysemy.Final                 ( Final )
+import           Polysemy.Output                ( Output
+                                                , output
+                                                , runOutputList
+                                                )
+import           System.IO                      ( Handle
+                                                , hPutStrLn
+                                                )
+
+import           HST.Effect.Cancel              ( Cancel
+                                                , cancel
+                                                , runCancel
+                                                )
+
+-------------------------------------------------------------------------------
+-- Messages                                                                  --
+-------------------------------------------------------------------------------
+
+-- | The severity of a 'Message'.
+data Severity = Internal | Error | Warning | Info | Debug
+  deriving (Show, Eq)
+
+-- | A messages that can be 'report'ed.
+data Message = Message { msgSeverity :: Severity, msgText :: String }
+
+-- TODO Add @Pretty@ instance for messages.
+showPrettyMessage :: Message -> String
+showPrettyMessage (Message severity msg) = show severity ++ ": " ++ msg
+
+-------------------------------------------------------------------------------
+-- Effect and Actions                                                        --
+-------------------------------------------------------------------------------
+
+-- | An effect capable of reporting 'Message's (e.g., errors and warnings).
+--
+--   It is distinguished between fatal and non-fatal messages. A fatal message
+--   is usually an error that cannot be recovered from.
+data Report m a where
+  Report ::Message -> Report m ()
+  ReportFatal ::Message -> Report m a
```

Comment:
Spaces?

Reply:
Even though Brittany does not format data type declarations, it removes spaces from constructors in GADTs :disappointed: 
Adding `-XGADTs` to Brittanies configuration file didn't work either. Floskell on the other hand formats them correctly. Since we have planed to switch to Floskell anyway, I've opened an issue on the `guidelines` repository that describes the migration process from Brittany to Floskell (see <https://github.com/FreeProving/guidelines/issues/5>).